vnext/ReactUWP/Views/TextInputViewManager.cpp

```diff
@@ -351,6 +364,36 @@ void TextInputShadowNode::registerEvents() {
   }
 }
 
+winrt::Shape TextInputShadowNode::FindCaret(winrt::DependencyObject element) {
+  if (element == nullptr)
+    return nullptr;
+
+  if (auto shape = element.try_as<winrt::Shape>())
+    return shape;
+
+  int childrenCount = winrt::VisualTreeHelper::GetChildrenCount(element);
+  for (int i = 0; i < childrenCount; i++) {
+    auto result = FindCaret(winrt::VisualTreeHelper::GetChild(element, i));
+    if (result != nullptr)
+      return result;
+  }
+
+  return nullptr;
+}
+
+// hacking solution to hide the caret
+void TextInputShadowNode::UpdateCaret() {
+  auto control = GetView().as<winrt::Control>();
+  auto caret = FindCaret(control);
+
+  if (caret != nullptr && m_hideCaret) {
```

Comment:
nit:
  if (m_hideCaret && caret) {

Reply:
possible the best way is 
if (m_hideCaret) {
   auto control = xx;
  if ( auto caret = yy) {
     //dothings
  }
}

---
In reply to: [334112433](https://github.com/microsoft/react-native-windows/pull/3397#discussion_r334112433) [](ancestors = 334112433)